invenio_rdm_records/marshmallow/json.py

```diff
@@ -36,34 +42,235 @@ class PersonIdsSchemaV1(StrictKeysMixin):
 class ContributorSchemaV1(StrictKeysMixin):
     """Contributor schema."""
 
+    ROLES = [
+        "ContactPerson",
+        "Researcher",
+        "Other"
+    ]
+
     ids = fields.Nested(PersonIdsSchemaV1, many=True)
     name = SanitizedUnicode(required=True)
     role = SanitizedUnicode()
     affiliations = fields.List(SanitizedUnicode())
     email = fields.Email()
 
+    @validates('role')
+    def validate_role(self, value):
+        """Validate that the role is one of the allowed ones."""
+        if value not in self.ROLES:
+            raise ValidationError(
+                _('Invalid role. Not one of {allowed}.'.format(
+                                                        allowed=self.ROLES)),
+                field_names=['contributor']
+            )
+
+
+class ResourceTypeSchemaV1(StrictKeysMixin):
+    """Resource type schema."""
+
+    type = fields.Str(
+        required=True,
+        error_messages=dict(
+            required=_('Type must be specified.')
+        ),
+    )
+    subtype = fields.Str()
+    openaire_subtype = fields.Str()
+    title = fields.Method('get_title', dump_only=True)
+
+    def get_title(self, obj):
+        """Get title."""
+        obj = ObjectType.get_by_dict(obj)
+        return obj['title']['en'] if obj else missing
+
+    @validates_schema
+    def validate_data(self, data):
+        """Validate resource type."""
+        obj = ObjectType.get_by_dict(data)
+        if obj is None:
+            raise ValidationError(_('Invalid resource type.'))
+
+    def dump_openaire_type(self, obj):
+        """Get OpenAIRE subtype."""
+        acc = obj.get('access_right')
+        if acc:
+            return AccessRight.as_category(acc)
+        return missing
+
+
+class TitleSchemaV1(StrictKeysMixin):
+    """Schema for the additional title."""
+
+    TITLE_TYPES = [
+          "AlternativeTitle",
+          "Subtitle",
+          "TranslatedTitle",
+          "Other"
+    ]
+
+    title = SanitizedUnicode(required=True, validate=validate.Length(min=3))
+    title_type = SanitizedUnicode()
+    lang = SanitizedUnicode()
+
+    @validates('lang')
+    def validate_language(self, value):
+        """Validate that language is ISO 639-3 value."""
+        validate_iso639_3(value)
+
+    @validates('title_type')
+    def validate_title_type(self, value):
+        """Validate that the title type is one of the allowed ones."""
+        if value not in self.TITLE_TYPES:
+            raise ValidationError(
+                _('Invalid title type. Not one of {allowed}.'.format(
+                                                    allowed=self.TITLE_TYPES)),
+                field_names=['additional_titles']
+            )
```

Comment:
Would this pattern be enough for "enums": 

`title_type = SanitizedUnicode(validate=validate.oneOf(TITLE_TYPES))` ?

Reply:
Yes, I added also `error` to be able to use the custom message with i18 babel (`_`) that was there and might be helpful later on. 

Nice catch! thanks!